src/main/java/net/fabricmc/loom/configuration/mods/MixinDetector.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2023 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.configuration.mods;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+import net.fabricmc.loom.LoomGradlePlugin;
+import net.fabricmc.loom.util.FileSystemUtil;
+import net.fabricmc.loom.util.fmj.FabricModJson;
+import net.fabricmc.loom.util.fmj.FabricModJsonFactory;
+
+public final class MixinDetector {
+	private static final Logger LOGGER = Logging.getLogger(MixinDetector.class);
+
+	public static boolean hasMixinsWithoutRefmap(Path modJar) throws IOException {
+		final List<String> mixinConfigs = getMixinConfigs(modJar);
+
+		if (!mixinConfigs.isEmpty()) {
+			try (FileSystemUtil.Delegate fs = FileSystemUtil.getJarFileSystem(modJar)) {
+				for (String mixinConfig : mixinConfigs) {
+					final Path configPath = fs.getPath(mixinConfig);
+
+					if (Files.exists(configPath)) {
```

Comment:
Nit: remove some of the indents, with a contiune, no need to wrap the whole thing in this if. Could maybe move the bellow code into its own method as well.